core/src/main/kotlin/configuration.kt

```diff
@@ -269,4 +270,12 @@ fun ExternalDocumentationLink(
 ): ExternalDocumentationLinkImpl =
     ExternalDocumentationLink(url.let(::URL), packageListUrl?.let(::URL))
 
+interface Option
+interface FeatureOption : Option {
+    val name: String
+}
+
+fun DokkaConfiguration.containsOption(option: FeatureOption) =
+    extraOptions.firstOrNull { it.indexOf("-XX") == 0 && option.name == it.substring(3) } != null
```

Comment:
- It does not limit users to implementing their own functions. Everyone can define an overload 
`fun DokkaConfiguration.containsOption(option: Option)`
- Besides, we must not encourage users to use the extra options instead of a standard configuration.
- >  With object, it's also kind of verbose:

Anyway, you can use your own short implementation:
```kt
    open class FeatureOptionImpl(override val name: String): FeatureOption
    object MyFeatureOption: FeatureOptionImpl("my-feature-opt")
```   
- > Necessary -XX start of the option is also a tight constraint - developers might have their own convention of option naming within the company, or would just like to do something different with meaning behind it.

See my PR comments above. 
- > Speaking of KGP, it looks like that's exactly the way it's implemented for `freeCompilerArgs`: no type safety, and consumers decide themselves how they want to check if a value is present.

KGP does not use arguments, it only calls a compiler with them.
Although, you can have a look under the hook of the compiler.  
https://github.com/JetBrains/kotlin/blob/master/compiler/cli/cli-common/src/org/jetbrains/kotlin/cli/common/arguments/InternalCompilerArgument.kt
https://github.com/JetBrains/kotlin/blob/master/compiler/util/src/org/jetbrains/kotlin/config/LanguageVersionSettings.kt

Reply:
At the moment, `Option` / `FeatureOption` API is very limiting and specific. If the users have to implement everything themselves, I fail to see why it needs to be public. Especially given that `Option` is not used anywhere as type.

Reply:
Can you give me the user cases you expect from extra options? 
I agree that in some cases `key=value` can be needed, but it might be introduced later via expanding the current API.
In my opinion, all cases of using extra options should be relevant to experimental features that we do not use want to use in a public configuration of a plugin. In other cases, options from a plugin configuration should be used.

Reply:
So the extra options are postponed until we have [more cases of using them](https://github.com/Kotlin/dokka/issues/2776).